schema/configuration-structure.asn

```diff
@@ -34,25 +34,161 @@ BEGIN
       }
    }
 
+   Version ::= SEQUENCE {
+      major	INTEGER ( 0 | 1 | ... ),
+      minor	INTEGER
+   }
+
    Hart ::= SEQUENCE {
       hartId   FlexibleRange,
 
-      -- Extensions that will not be common on small platforms should go here.
-      -- When these extensions are not used they take no space at all.
-      extensions SEQUENCE OF CHOICE {
-         debug Debug,
-         custom SEQUENCE OF Custom,
-         zjpm Zjpm,
-         zk Zk,
+      -- Base Integer ISA indicating the supported address space size in bits
+      baseIsa CHOICE {
+         rv32I NULL,
+         rv64I NULL,
+         ...
+      }, -- baseISA
+      baseIsaVersion Version,
+
+      stdUnprivExtensions SEQUENCE OF CHOICE {
+         multiplication CHOICE {
+            m     M     OPTIONAL, -- M: Integer Multiplication and Division
+            zmmul Zmmul OPTIONAL, -- Zmmul: Multiplication Subset of the M extension
+            ...
+         } OPTIONAL, -- multiplication
+         a A OPTIONAL, -- A: Atomic Memory Operations
+         fpDoublePrecision CHOICE {
+            d D,          -- D: Floating-Point, Double-Precision.  Implies F
+            zdinx  Zdinx, -- Zdinx: Floating-Point, Double-Precision in Integer Registers X.  Implies Zfinx
+            ...
+         } OPTIONAL, -- fpDoublePrecision
+         fpSinglePrecision CHOICE {
+            f F OPTIONAL, -- F: Floating-Point, Single-Precision.  Implies Zicsr
+            zfinx  Zfinx, -- Zfinx: Floating-Point, Single-Precision in Integer Registers X.
+            ...
+         } OPTIONAL, -- fpSinglePrecision
+         q Q OPTIONAL, -- Q: Quad-Precision Floating-Point.  Implies D
```

Comment:
I don't think you ever need a value inside a CHOICE to be OPTIONAL. If the thing isn't implement, simply don't choose it at all.

Reply:
Right, the reason it was there it was because before our meeting of Nov/7, stdUnprivExtensions was a SEQUENCE, not a SEQUENCE OF CHOICE.  I have deleted all OPTIONAL clauses from all CHOICE structures.  Thanks